### PR TITLE
Stop planning unnecessary property projections

### LINKED ISSUE #580: Avoid projecting property expressions where possible

### PR DESCRIPTION
**Pull request by s1ck**
* fixes #580

### CODE REVIEW

File: spark-cypher-testing/src/test/scala/org/opencypher/spark/impl/acceptance/PredicateBehaviour.scala [hunk 1]
```diff
@@ -35,6 +35,14 @@ import org.scalatest.DoNotDiscover
 @DoNotDiscover
 class PredicateBehaviour extends CAPSTestSuite with DefaultGraphInit {
 
+  it("can evaluate predicates on non-existing properties") {
```
Comment (Mats-SX) [inline spark-cypher-testing/src/test/scala/org/opencypher/spark/impl/acceptance/PredicateBehaviour.scala]: Was this test case red before the fix?

Comment (s1ck) [reply spark-cypher-testing/src/test/scala/org/opencypher/spark/impl/acceptance/PredicateBehaviour.scala]: No, it's an additional test that verifies we correctly evaluate predicate expressions based on non-existing properties.

Comment (s1ck) [reply spark-cypher-testing/src/test/scala/org/opencypher/spark/impl/acceptance/PredicateBehaviour.scala]: Before we have just one spark-cypher-test (`can evaluate coalesce on non-existing expressions`) that implicitly tests for that.

Comment (Mats-SX) [reply spark-cypher-testing/src/test/scala/org/opencypher/spark/impl/acceptance/PredicateBehaviour.scala]: Very nice.

**Pull request merged**